### ISSUE TITLE
MI1S - trigger alarm if heart rate too high or low?

### ISSUE THREAD

**Issue by rgh36167** [Freeyourgadget/Gadgetbridge#372]
also would be nice to email or otherwise send those

**Comment by rgh36167** [comment]
Noticed a discrepancy in https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Information and real life regarding the version of  MI1S HR firmware: the wiki page says the recommended versions for MI1S are 4.15.12.10 Mi Band + 1.3.74.76 HR and were contained in Mifit apks 2.0.1, 2.0.4, 2.0.5.

I extracted and installed the firmware from 2.0.5 on the band, now Gadgetbridge now reports the firmware versions as 4.15.12.10 + 1.3.74.64.

The md5sums however do match, maybe a typo in the wiki page? 

The

**Comment by rgh36167** [comment]
Btw a big compliment - got the device yesterday, installed app from f-droid and it worked within minutes. Needed a different firmware for the HR and even that went smoothly. Working with a Moto G 1st generation.

However - so far I can get the heart rate only in the debug settings - the "Activity and Sleep" graph seems like it should display it but does not? Do I need to build my own version to enable it?

**Comment by d-p** [comment]
The title of the issue does not match the comments, anyway...

I made the last modifications to the page.

I double checked the info I added but not on those already present.

I confirm that also that in my mi band the version numbers are
"4.15.12.10 + 1.3.74.64".  The version 1.3.74.76 was added by
@cpfeiffer, that maybe can confirm or not the number.

I used https://github.com/danielrheinbay/mi-band-firmware-analyse that
unfortunately does not print the hr firmware version.

Also the table formatting seems good in the preview while editing but
in the show the first colum division line does not appear.

Maybe is also the case to integrate what is in:
https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Update#known-firmwares-md5

and then remove the page to avoid partial duplication and confusion.
Let me know I will take care to do the merge.

**Comment by d-p** [comment]
> Also the table formatting seems good in the preview while editing but
> in the show the first colum division line does not appear.

Oh, just found out that is a browser "feature" that depend on scaling the page...

**Comment by cpfeiffer** [comment]
Thanks, I've fixed the firmware version in the wiki, that was indeed a typo. I also marked 4.16.4.22 as working. I think someone had mentioned problems with it, but I'm running it for weeks without any problem.

**Comment by cpfeiffer** [comment]
@d-p right, the md5 information from https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Update#known-firmwares-md5 should be removed in favor of https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Information
IMHO the plain version numbers + md5sum don't make much sense anymore anyway.

A link to the other page would be more helpful.

@rgh36167 thanks for the kind words. The Activity and Sleep chart only displays (sleep) hr data if you enabled that in the settings. You can get live hr values if you swipe one more to the right.

Besides, @dp is right, all this is quite offtopic in this issue.
